lib/discovery.js

```diff
@@ -501,4 +343,12 @@ function mixinDiscovery(DB2, db2) {
     // }
     // return undefined;
   };
+
+  DB2.prototype.setDefaultOptions = function(options) {
```

Comment:
@raymondfeng Sorry on a second thought, I think we'd better still keep the function in base connector throwing error, see https://github.com/strongloop/loopback-connector/blob/master/lib/sql.js#L1795
I add a new comment in the parent refactor story, let's keep the 2nd round discussion in https://github.com/strongloop/loopback-connector/issues/15#issuecomment-283163453